### PR TITLE
ACCUMULO-4800 Cache parsing of iterator config

### PR DESCRIPTION
**Pull request by keith-turner**

### CODE REVIEW

File: server/base/src/main/java/org/apache/accumulo/server/conf/TableConfiguration.java [hunk 1]
```diff
@@ -144,4 +163,50 @@ public String toString() {
...
+      return tableOpts;
+    }
+
+    public String getContext() {
+      return context;
+    }
+  }
+
+  public ParsedIteratorConfig getParsedIteratorConfig(IteratorScope scope) {
+    long count = getUpdateCount();
```
Comment (mikewalch) [inline server/base/src/main/java/org/apache/accumulo/server/conf/TableConfiguration.java]: I see that the iterator config will change if this count changes.  When does this count change?

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/ScanDataSource.java [hunk 2]
```diff
@@ -186,12 +188,28 @@ public boolean isCurrent() {
...
     if (!loadIters) {
       return visFilter;
-    } else if (null == options.getClassLoaderContext()) {
-      return iterEnv.getTopLevelIterator(IteratorUtil.loadIterators(IteratorScope.scan, visFilter, tablet.getExtent(), tablet.getTableConfiguration(),
-          options.getSsiList(), options.getSsio(), iterEnv));
     } else {
-      return iterEnv.getTopLevelIterator(IteratorUtil.loadIterators(IteratorScope.scan, visFilter, tablet.getExtent(), tablet.getTableConfiguration(),
-          options.getSsiList(), options.getSsio(), iterEnv, true, options.getClassLoaderContext()));
+      List<IterInfo> iterInfos;
+      Map<String,Map<String,String>> iterOpts;
```
Comment (mikewalch) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/ScanDataSource.java]: Can you describe what is going on in this block?

File: server/base/src/main/java/org/apache/accumulo/server/conf/TableConfiguration.java (hunk 1 above)
Comment (keith-turner) [reply server/base/src/main/java/org/apache/accumulo/server/conf/TableConfiguration.java]: I added some comments to the AccumuloConfiguration super class.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/ScanDataSource.java (hunk 2 above)
Comment (keith-turner) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/ScanDataSource.java]: I added some comments

**Pull request merged**